src/main/java/io/reactivex/internal/operators/observable/ObservableInternalHelper.java

```diff
@@ -319,7 +320,7 @@ public boolean test(Notification<Object> t) throws Exception {
         @Override
         public Observable<R> apply(T t) throws Exception {
             return RxJavaPlugins.onAssembly(new SingleToObservable<R>(
-                ObjectHelper.requireNonNull(mapper.apply(t), "The mapper returned a null value")));
+                ObjectHelper.requireNonNull(mapper.apply(t), "The mapper returned a null SingleSource")));
```

Comment:
Just open the collapsed section and see what mapper is: `final Function<? super T, ? extends SingleSource<? extends R>> mapper;`